Pokedex/ViewController.swift

```diff
@@ -0,0 +1,345 @@
+//
+//  ViewController.swift
+//  Pokedex
+//
+//  Created by SAMEER SURESH on 9/25/16.
+//  Copyright © 2016 trainingprogram. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    let pokeGen = PokemonGenerator.getPokemonArray()
+    var pokemon : [Pokemon]! = []
+    var nameButton : UIButton!
+    var numButton : UIButton!
+    var attackButton : UIButton!
+    var healthButton : UIButton!
+    var defenseButton : UIButton!
+    var randomButton : UIButton!
+    var collectionView : UICollectionView!
+    var selectCategoryLabel : UILabel!
+    var searchBar : UITextField!
+    var byTypesLabel : UILabel!
+    var searchByCategoriesButton : UIButton!
+    var searchByTypesButton : UIButton!
+    var pressed : UIButton!
+    var searchedPokemon : [Pokemon] = []
+    var noPokemonToBeSearched : Bool!
+    let pokemonTypeNames : [String]! = ["Bug", "Dark", "Dragon", "Electric", "Fairy", "Fighting", "Fire", "Flying", "Ghost", "Grass", "Ground", "Ice", "Normal", "Poison", "Psychic", "Rock", "Steel", "Water"]
+    var selectedTypesNames : [String]! = []
```

Comment:
Most of the above variables do not need to be unwrapped. Only use the (!) when you have to. You can force unwrap with the (!) when the variable is instantiated somewhere else, but you know it will exist.